### ISSUE TITLE
VS2022 support?

### ISSUE THREAD

**Issue by anupshah** [Wumpf/VSSolutionColor#22]
Hi, thanks for this useful extension.

Any plans to update to support VS2022?

**Comment by Wumpf** [comment]
No, not using this or VS in general much anymore myself. But if someone wants to take it over we can make this happen

**Comment by ovidiubuligan** [comment]
please make it happen , we need this . @Wumpf  what are you using instead of VS ?

**Comment by Roozi489** [comment]
Hey, I made the extension work with VS2022. It was a quick hack so it only works with 64bit VS2022.
I did not test it much, but it seems to work fine.
It also contains a couple of changes. If automatic color is used it searches for color names to use in solution path. If if doesn't find any color it uses a hash of the path so that the color stays the same even when solution is recreated.
Hope this is OK @Wumpf 
[SolutionColor_fromPath_vs2022.zip](https://github.com/Wumpf/VSSolutionColor/files/7521737/SolutionColor_fromPath_vs2022.zip)

**Comment by Wumpf** [comment]
Thanks! Can you create a pr?

**Comment by Wumpf** [comment]
>  It was a quick hack so it only works with 64bit VS2022.

ah well nvm actually, wouldn't want to land that obviously ;)

**Comment by Roozi489** [comment]
I can clean it up and create a PR if you want. It was just a very minor change.

**Comment by Caedendi** [comment]
@Roozi489 Please make a PR :) thx for your work.

**Comment by ovidiubuligan** [comment]
Yes, +1

**Comment by Caedendi** [comment]
@Roozi489 any chance you could make a PR?

**Comment by Roozi489** [comment]
Sorry, yeah, I'm still planning to.
I just didn't know how much work is it to make the extension work with both VS2019 and VS2022. The code change itself is trivial, but you need to have at least 2 separate projects. MS [recommends](https://docs.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022) moving the code into Shared project and creating one project for VS2019 (and older) and one for VS2022.

**Comment by twilly86** [comment]
Yes, +1

**Comment by RODevVro** [comment]
Hello, I added the support for VS 2022 available on this fork [here](https://github.com/RODevVro/VSSolutionColor). You just have to download the solution, build in Release and you will find the .vsix extension in the bin/ folder. Separate the project into a shared project as MS [recommends](https://docs.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022).

I don't do PR yet since I can't get the extension to work under VS 2019. The extension installs but I don't see the commands accessible from the VS menus. There is still some work to do but the fork allows to use the plugin under VS 2022.

**Comment by dviry** [comment]
Just for those looking for an official VS2022 supported plugin that does the same: 

https://devblogs.microsoft.com/visualstudio/differentiating-visual-studio-instances/

**Comment by anupshah** [comment]
Thanks @dviry - I had seen that myself and started using that.

I appreciate that extension development has become a bit more complex if you need to support both VS 2022 and VS 2019.

For me, I am happy to use that other extension for now, so if you want to close this issue that is fine with me, though others may still prefer this extension and want to keep this open?

**Comment by rajniszp** [comment]
The plugin mentioned above is "Solution Colors" by Mads Kristensen
https://marketplace.visualstudio.com/items?itemName=MadsKristensen.SolutionColors

Also, in VS 2026 it is build in (Options -> Environment -> Visual Experience -> Color theme with solution scope selected at the top bar: https://www.youtube.com/watch?v=jow_IUKNZkU&t=596s )
